Userland/Libraries/LibWebSocket/Impl/WebSocketImpl.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021, Dex♪ <dexes.ttp@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWebSocket/Impl/WebSocketImpl.h>
+
+namespace WebSocket {
+
+WebSocketImpl::WebSocketImpl(Core::Object* parent)
+    : Object(parent)
+{
+}
+
+WebSocketImpl::~WebSocketImpl()
+{
+}
+
+void WebSocketImpl::connect(ConnectionInfo const& connection_info)
```

Comment:
I suppose we cannot make this a static function unless we get the ability to yield to the current event loop sometime in the future...

Reply:
We can copy the thing `TLSv12::connect` does, though it seems like you hate that

Reply:
I'd say let's keep this for a future change, this PR is already large enough without the extra WebSocket API change